pymongo/srv_resolver.py

```diff
@@ -111,9 +113,12 @@ def _get_srv_response_and_hosts(self, encapsulate_errors):
                 raise ConfigurationError("Invalid SRV host: %s" % (node[0],))
             if self.__plist != nlist:
                 raise ConfigurationError("Invalid SRV host: %s" % (node[0],))
-
+        if(self.__srv_max_hosts != 0):
+            ret = list(zip(results, nodes))
+            random.shuffle(ret)
```

Comment:
Why isn't this using random.sample? Ideally we can refactor this logic into a single place that's shared between the initial discovery phase and _updated_topology_description_srv_polling.

Reply:
So actually I think we want to use the shuffle function in both places because the shuffle function basically does exactly what the go driver implements (https://stackoverflow.com/questions/12798363/what-is-the-difference-between-random-sample-and-random-shuffle-in-python), and I also think it looks cleaner and is easier to read. But I wanted to make sure that this code made sense to you before changing both of them.